WorkloadManagementSystem/Executor/JobScheduling.py

```diff
@@ -222,16 +222,12 @@ def optimizeJob( self, jid, jobState ):
     return self.__setJobSite( jobState, stageSites )
 
   def __applySiteFilter( self, sites, active = False, banned = False ):
-    filtered = list( sites )
-    if active:
-      for site in sites:
-        if site not in active:
-          filtered.remove( site )
-    if banned:
-      for site in banned:
-        if site in filtered:
-          filtered.remove( site )
-    return filtered
+    filtered = set( sites )
+    if active and isinstance( active, ( list, set, dict ) ):
+      filtered &= set( active )
+    if banned and isinstance( banned, ( list, set, dict ) ):
+      filtered -= set( banned )
+    return list( filtered )
 
```

Comment:
I know you just re-wrote it in a better way, in fact as I said it was not fully correct even before.

My change is done to cover a very special case indeed, that is for example when user requires [SiteA], but SiteA is not in the list of "allowed" nor "removed".  Basically it is for https://github.com/fstagni/DIRAC/blob/02f90336e6bd2626338fbec2f80309b6ecdfa9a8/WorkloadManagementSystem/Executor/test/Test_Executor.py#L70